internal/parser/scanner/states.go

```diff
@@ -192,37 +383,59 @@ var keywordsWithT map[string]token.Type = map[string]token.Type{
 	"TRIGGER":     token.KeywordTrigger,
 }
 
-var keywordsWithU map[string]token.Type = map[string]token.Type{
+var keywordsWithU = []string{
+	"UNBOUNDED",
+	"UNION",
+	"UNIQUE",
+	"UPDATE",
+	"USING",
+}
+
+var keywordMapWithU map[string]token.Type = map[string]token.Type{
 	"UNBOUNDED": token.KeywordUnbounded,
 	"UNION":     token.KeywordUnion,
 	"UNIQUE":    token.KeywordUnique,
 	"UPDATE":    token.KeywordUpdate,
 	"USING":     token.KeywordUsing,
 }
 
-var keywordsWithV map[string]token.Type = map[string]token.Type{
+var keywordsWithV = []string{
+	"VACUUM",
+	"VALUES",
+	"VIEW",
+	"VIRTUAL",
+}
+
+var keywordMapWithV map[string]token.Type = map[string]token.Type{
 	"VACUUM":  token.KeywordVacuum,
 	"VALUES":  token.KeywordValues,
 	"VIEW":    token.KeywordView,
 	"VIRTUAL": token.KeywordVirtual,
 }
 
-var keywordsWithW map[string]token.Type = map[string]token.Type{
+var keywordsWithW = []string{
+	"WHEN",
+	"WHERE",
+	"WINDOW",
+	"WITH",
+	"WITHOUT",
+}
+
+var keywordMapWithW map[string]token.Type = map[string]token.Type{
 	"WHEN":    token.KeywordWhen,
 	"WHERE":   token.KeywordWhere,
 	"WINDOW":  token.KeywordWindow,
 	"WITH":    token.KeywordWith,
 	"WITHOUT": token.KeywordWithout,
 }
 
-// func scanSpace(s *scanner) token.Token {
-// 	s.accept(matcher.String(" "))
-// 	return createToken(s.line, s.col, s.start, s.pos, token.SQLSpecialCharacter, string(s.input[s.start:s.pos]), s)
-// }
+func scanSpace(s *scanner) {
+	s.accept(matcher.String(" "))
```

Comment:
Don't create matchers inline, it is expensive and unneccessary.

Reply:
Removed.

Reply:
Changes committed? It still shows up here.